llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -5760,6 +5760,17 @@ bool SelectionDAG::isKnownNeverNaN(SDValue Op, const APInt &DemandedElts,
         return false;
     return true;
   }
+  case ISD::AssertNoFPClass: {
+    SDValue SDNoFPClass = Op.getOperand(1);
+    assert(isa<ConstantSDNode>(SDNoFPClass) && "NoFPClass is not Constant");
+    FPClassTest NoFPClass = static_cast<FPClassTest>(
+        dyn_cast<ConstantSDNode>(SDNoFPClass)->getZExtValue());
```

Comment:
```suggestion
    FPClassTest NoFPClass = static_cast<FPClassTest>(Op.getConstantOperandVal(1));
```

cast instead of isa + dyn_cast, but here we have a helper